packages/window-size-tool/src/engine.js

```diff
@@ -109,21 +112,18 @@ function checkLinkWidth(pageNode, rule, verbose = true) {
  * @param {boolean} verbose Will log debug messages if true (true by default).
  * @returns Whether the DOM node meets the rules criteria or not.
  */
-function checkRule(node, rule, verbose = true, domainCheckPass = false) {
+async function checkRule(node, filename, rule, verbose = true, domainsCheckEnabledForAllRules = true) {
   let pass = false;
+  let domainsCheckPass = false;
 
   if (!node) {
     throw Error("You must supply a node");
   } else if (!rule) {
     throw Error("You must supply a rules object");
   }
-  if (rule.containsAllowedDomainsOnly !== undefined && rule.containsAllowedDomainsOnly === true) {
-    // if containsAllowedDomainsOnly set and the check has failed return with failure, ohterwise 
-    // check the other rules
-    if (domainCheckPass===false) { 
-      return;
-    }
-  }
+
+  const domainsCheckForSpecificRule = rule && rule.containsAllowedDomainsOnly;
+  const executeDomainCheck = domainsCheckForSpecificRule != undefined ? domainsCheckForSpecificRule : domainsCheckEnabledForAllRules;
```

Comment:
It might not matter but won't `domainsCheckForSpecificRule != undefined ?` always return true because you've defined it on the previous line to a boolean? Maybe I'm wrong.